cli/azd/pkg/infra/provisioning/bicep/bicep_provider_test.go

```diff
@@ -555,11 +555,25 @@ func prepareDestroyMocks(mockContext *mocks.MockContext) {
 	})
 
 	// Delete deployment
-	mockContext.CommandRunner.When(func(args exec.RunArgs, command string) bool {
-		return strings.Contains(command, "az deployment sub delete")
-	}).Respond(exec.RunResult{
-		Stdout: "",
-		Stderr: "",
+	mockPollingUrl := "https://url-to-poll.net/keep-deleting"
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.Method == http.MethodDelete &&
+			strings.Contains(
+				request.URL.Path, "/subscriptions/SUBSCRIPTION_ID/providers/Microsoft.Resources/deployments/test-env")
+	}).RespondFn(func(request *http.Request) (*http.Response, error) {
+		response, err := mocks.CreateEmptyHttpResponse(request, 202)
+		response.Header.Add("Operation-Location", mockPollingUrl)
+		return response, err
+	})
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.Method == http.MethodGet &&
+			strings.Contains(
+				request.URL.String(), mockPollingUrl)
+	}).RespondFn(func(request *http.Request) (*http.Response, error) {
+		type theBody struct {
+			Status string `json:"status,omitempty"`
+		}
+		return mocks.CreateHttpResponseWithBody(request, 204, theBody{Status: "Succeeded"})
```

Comment:
The response code here in the mock response is a little surprising to me, 204 is "No Content", I would have expected 200 given that we are returning a body...

Reply:
One of the weirdest APIs I've ever seen.  I am totally with you about the `200`, but look:
https://learn.microsoft.com/en-us/rest/api/resources/deployments/delete-at-subscription-scope

They return 202 during the polling plus a location header to polling.  On operation completed, they return 204 (and the SDK stills expects a body with the status.

This is yet, one more time where I can't wait for record&playback.  :)

Reply:
Yes - that is very strange.  I learn something new every day!  Thanks for the explanation.